src/forms/labels/field-set-legend.js

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import { convertNameToLabel } from '../helpers'
+
+/**
+ *
+ * A legend representing a caption for the content of its parent field set element
+ *
+ * This component must be used as a direct child and the only legend of the <fieldset> element that groups related controls
+ * 
+ *
+ * The text of the legend is set using the following rules:
+ * - If the `label` prop is set to `false`, the legend is hidden visually
```

Comment:
Let's specify that it's hidden visually via a class and that its the consumer's responsibility to make sure their styling rules respect the `visually-hidden` class